### ISSUE TITLE
Support default for enum fields

### ISSUE THREAD

**Issue by pjoe** [aws-amplify/amplify-data#452]
### Describe the feature you'd like to request

The enum type doesn't currently support defining default value, e.g. this will not work:
```ts
a.enum(["a", "b"]).default("a")
```

### Describe the solution you'd like

Code like above works, just like with other field types.

### Describe alternatives you've considered

Had to use another field type (boolean or integer)

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change

**Comment by bryan-ds** [comment]
Hey team, this is currently marked as a [supported feature here](https://docs.amplify.aws/react/build-a-backend/data/data-modeling/add-fields/):

> You can use the .default(...) modifier to specify a default value for optional [scalar type fields and **enums**](https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html). The .default(...) modifier is not available for custom types, arrays, or relationships.

This is an important feature to add. Can we please get an update on this?

**Comment by stocaaro** [comment]
@bryan-ds, Good catch that the docs capture this behavior as supported when it is not supported yet. I have a change out to fix the docs and this is in our feature request list to support in the future.

@licarpen, When this is supported it is likely to be supported as a modifier on `a.ref('myEnum').default('value')` first vs directly on the `a.enum` builder inline. Because the `a.enum` builder is used both inline as well as in the top level to define enum's that are shared across multiple interfaces, it is awkward to support the `.required`, `.array` and now `.default` modifiers which make sense inline, but aren't applicable used at the top level definition.

Thanks,
Aaron

**Comment by Simone319** [comment]
@speedhawk21 thanks for your interest in this feature request! We will post an update here when this feature is implemented.
